Shared/sdk/SharedUtil.File.hpp

```diff
@@ -1052,9 +1066,60 @@ int SharedUtil::File::Delete(const char* szFilename)
 
 int SharedUtil::File::Rename(const char* szOldFilename, const char* szNewFilename)
 {
-#ifdef WIN32
+#ifdef _WIN32
     return _wrename(FromUTF8(szOldFilename), FromUTF8(szNewFilename));
 #else
     return rename(szOldFilename, szNewFilename);
 #endif
 }
+
+std::vector<std::string> SharedUtil::ListDir(const char* szPath) noexcept
+{
+    std::vector<std::string> entries;
+#if _HAS_CXX17
+    namespace fs = std::filesystem;
+    if (!DirectoryExists(szPath))
+        return {};
+
+    try
+    {
+        for (const auto& entry : fs::directory_iterator(szPath))
+        {
+            if (entry.is_regular_file() || entry.is_directory())
+                entries.push_back(entry.path().filename().string());
+        }
+    }
+    catch (const fs::filesystem_error& exc) {}
+#else
+    #ifdef _WIN32
+    std::string search_path = szPath;
+    if (search_path.back() != '/')
```

Comment:
When a string is empty method `back` has undefined behaviour. I would recommend to add a check `!search_path.empty()`.